### DISCUSSION TITLE
browser extension of Firefly

### DISCUSSION THREAD

**Discussion by aayushpoudel-vizio** [iotaledger/firefly#1694] [Features / Enhancements]
Motivation:
As ISCP is in beta and we are about to release Shimmer, Assembly
we need Metamask like wallet where dapp can connect. For EVM contracts we can use Metamask but for WASM contracts we need browser wallet that support iota address, also all sort of JS libraries for developer to interact with WASM SC

**Comment by LauGau** [comment]
Yes I think it’s more convenient with a browser extension. But I definitely LOVE the Firefly desktop app so I think the best move will be to have both and keep the most critical tasks to the desktop app. (So the browser extension will act like a relay).

**Comment by r-c-k** [comment]
In development: [youtu.be/HPHYwfkUKro?t=5437](https://youtu.be/HPHYwfkUKro?t=5437)
